packages/braid-design-system/lib/hooks/useIcon/index.ts

```diff
@@ -57,9 +59,11 @@ export default (
   const headingContext = useContext(HeadingContext);
   const inheritedTone =
```

Comment:
_(GitHub won't let me add a suggestion over deleted lines)_

With the `| null` change I suggested in `TextContext`, this can be simplified:

```ts
  const inheritedTone = textContext?.tone || 'neutral';
  const resolvedTone = typographyStyles.tone[tone || inheritedTone];
  const toneClass =
    tone || (!headingContext && !textContext?.tone) ? resolvedTone : undefined;
```

Unit tests and `tsc` passed locally.